clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -863,9 +865,14 @@ public long position(TopicPartition partition, Duration timeout) {
 
             Timer timer = time.timer(timeout);
             do {
-                SubscriptionState.FetchPosition position = subscriptions.validPosition(partition);
-                if (position != null)
-                    return position.offset;
+                // if cachedSubscriptionHasAllFetchPositions is false, there is a ResetOffsetEvent in queue
```

Comment:
This statement may be wrong. Even if there is no `ResetOffsetEvent`, the `cachedSubscriptionHasAllFetchPositions` still can be false, like consumer just initialization. Probably, we can change it like:

```java
/**
 * The {@link seekToBeginning} and {@link seekToEnd} update subscription state position in background.
 * When {@link cachedSubscriptionHasAllFetchPositions} is false, there may have {@link ResetOffsetEvent}, which means
 * current subscription valid position is not reliable. We should update fetch position in the background thread.
 */
```

Reply:
You are right, thanks.